### PR TITLE
Add an example app that uses extensions

### LINKED ISSUE #1377: Interprocess: shared container sample app
Test out interprocess pr in a sample app using shared containers.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1377.

\c @tgoyne @jpsim

### CODE REVIEW

File: examples/ios/objc/Extension/AppDelegate.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (tgoyne) [inline examples/ios/objc/Extension/AppDelegate.h]: all new files need a proper license header

File: examples/ios/objc/Extension/AppDelegate.m [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    return YES;
+}
+
+@end
+
+@implementation TickViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [RLMRealm setDefaultRealmPath:[[[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.io.realm.examples.extension"] URLByAppendingPathComponent:@"extension.realm"].path];
```
Comment (tgoyne) [inline examples/ios/objc/Extension/AppDelegate.m]: Shouldn't need to do this twice.

File: examples/ios/objc/RealmExamples.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -737,6 +1125,192 @@
...
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CODE_SIGN_ENTITLEMENTS = extension/extension.entitlements;
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"/Users/segiddins/Development/Realm/realm-cocoa/build/ios",
```
Comment (tgoyne) [inline examples/ios/objc/RealmExamples.xcodeproj/project.pbxproj]: path

File: examples/ios/objc/Extension/AppDelegate.m [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    [self tock];
+    [self updateLabel];
+}
+
+- (void)updateLabel {
+    [self.button setTitle:[NSString stringWithFormat:@"%ld", (long)self.tick.count] forState:UIControlStateNormal];
```
Comment (tgoyne) [inline examples/ios/objc/Extension/AppDelegate.m]: [self.button setTitle:@(self.tick.count).stringValue forState:UIControlStateNormal]

File: examples/ios/objc/Extension/AppDelegate.h (hunk 1 above)
Comment (segiddins) [inline examples/ios/objc/Extension/AppDelegate.h]: believe I hit all the new source files

File: Realm/Tests/PerformanceTests.m [hunk 5]
```diff
@@ -29,7 +29,7 @@ + (RLMPropertyAttributes)attributesForProperty:(__unused NSString *)propertyName
 }
 @end
 
-#if !DEBUG
+//#if !DEBUG
```
Comment (jpsim) [inline Realm/Tests/PerformanceTests.m]: This is temporary for CI, I hope?

File: examples/ios/objc/TodayExtension/TodayViewController.m [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    self.preferredContentSize = CGSizeMake(0, 200.0);
+    [RLMRealm setDefaultRealmPath:[[[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.io.realm.examples.extension"] URLByAppendingPathComponent:@"extension.realm"].path];
+    self.tick = [Tick allObjects].firstObject;
+    if (!self.tick) {
+        [[RLMRealm defaultRealm] transactionWithBlock:^{
+            self.tick = [Tick createInDefaultRealmWithObject:@[@"", @0]];
+        }];
+    }
+    self.notificationToken = [self.tick.realm addNotificationBlock:^(NSString *notification, RLMRealm *realm) {
+        if (self.tick.count % 19 == 0) {
```
Comment (jpsim) [inline examples/ios/objc/TodayExtension/TodayViewController.m]: could you add an explanatory comment here, and all other notification blocks doing something similar?

File: Realm/Tests/PerformanceTests.m (hunk 5 above)
Comment (segiddins) [inline Realm/Tests/PerformanceTests.m]: Ah dammit, I think I messed up the rebase. I'll fix.

**Comment by jpsim** [comment]
You should add these targets to `build.sh examples`, which would make sure they continue to build.

Adding comments explaining the rationale behind the modulo's would be nice, so we don't forget when we look at this again in a few months. It would also be nicer if the 13/17/19 values were kept in a single `Constants.h` file shared by all three targets.

Otherwise, :+1:

**Comment by segiddins** [comment]
The 13/17/19 are just three low prime numbers, so I don't see what use putting them in a Constants file will be. But I will add some comments.

**Comment by jpsim** [comment]
> The 13/17/19 are just three low prime numbers, so I don't see what use putting them in a Constants file will be.

Sure, doesn't matter.

**Comment by segiddins** [comment]
> You should add these targets to build.sh examples, which would make sure they continue to build.

Done.

File: Realm/Tests/PerformanceTests.m (hunk 5 above)
Comment (tgoyne) [inline Realm/Tests/PerformanceTests.m]: The `#if !DEBUG` makes them not show up in the Xcode UI, and this is rebased on a WIP branch that I need to be able to test.

Comment (segiddins) [inline Realm/Tests/PerformanceTests.m]: @tgoyne it should still be there, but I fixed the rebase

**Pull request merged**